lib/slavery/connection_holder.rb

```diff
@@ -4,19 +4,24 @@ class ConnectionHolder < ActiveRecord::Base
 
     class << self
       # for delayed activation
-      def activate
-        spec = ActiveRecord::Base.configurations[Slavery.spec_key]
+      def activate(spec_key = nil)
+        spec = ActiveRecord::Base.configurations[spec_key || Slavery.spec_key]
         raise Error.new('Slavery.spec_key invalid!') if spec.nil?
         establish_connection spec
       end
     end
   end
 
   class << self
-    def connection_holder
-      @connection_holder ||= begin
-        ConnectionHolder.activate
-        ConnectionHolder
+    def connection_holder(target,spec_key)
+      slave_connections[spec_key] ||= begin
+        klass = Class.new(Slavery::ConnectionHolder) do
+          self.abstract_class = true
+        end
+        klass_name = "SlaveryConnection#{target.to_s.camelize}"
+        Object.const_set(klass_name, klass) unless Object.const_defined?(klass_name)
```

Comment:
If const is already defined, it should just raise an explicit error rather than attempt `activate`.